File: server/src/main/java/gmetrica/enumeration/TipoProjeto.java
package gmetrica.enumeration;

public enum TipoProjeto {
	
	APURACAO_ESPECIAL ("Apuração Especial"),
	DESENVOLVIMENTO ("Desenvolvimento"),
	MANUTENCAO_ADAPTATIVA ("Manutenção Adaptativa"),
	MANUTENCAO_CORRETIVA ("Manutenção Corretiva"),
	MANUTENCAO_EVOLUTIVA ("Manutenção Evolutiva");
	
	private String descricao;
	
	private TipoProjeto(String descricao) {
		this.descricao = descricao;
	}

	public String getDescricao() {
		return descricao;
	}

	public static TipoProjeto fromDescricao(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(TipoProjeto valor: TipoProjeto.values()) {
			
			if(valor.getDescricao().equals(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
	
	public static TipoProjeto fromContains(String descricao) {
		if(descricao == null) {
			return null;
		}
		
		for(TipoProjeto valor: TipoProjeto.values()) {
			descricao = descricao.substring(0,1).toUpperCase().concat(descricao.substring(1));
			if(valor.getDescricao().contains(descricao)) {
				return valor;
			}
		}
		
		return null;
	}
}

File: client/src/app/enumeration/tipo-projeto.ts
import { SelectItem } from "primeng/components/common/selectitem";

export enum TipoProjeto {
    APURACAO_ESPECIAL = <any>"Apuração Especial",
    DESENVOLVIMENTO = <any>"Desenvolvimento",
    MANUTENCAO_ADAPTATIVA = <any>"Manutenção Adaptativa",
    MANUTENCAO_CORRETIVA = <any>"Manutenção Corretiva",
    MANUTENCAO_EVOLUTIVA = <any>"Manutenção Evolutiva"
}

export namespace TipoProjeto {
    export function getSelectItem(): any {
        var item: SelectItem[] = [];
        item.push({label:'Selecione', value:null});
        item.push({label:TipoProjeto.APURACAO_ESPECIAL.toString(), value:TipoProjeto[TipoProjeto.APURACAO_ESPECIAL]});
        item.push({label:TipoProjeto.DESENVOLVIMENTO.toString(), value:TipoProjeto[TipoProjeto.DESENVOLVIMENTO]});
        item.push({label:TipoProjeto.MANUTENCAO_ADAPTATIVA.toString(), value:TipoProjeto[TipoProjeto.MANUTENCAO_ADAPTATIVA]});
        item.push({label:TipoProjeto.MANUTENCAO_CORRETIVA.toString(), value:TipoProjeto[TipoProjeto.MANUTENCAO_CORRETIVA]});
        item.push({label:TipoProjeto.MANUTENCAO_EVOLUTIVA.toString(), value:TipoProjeto[TipoProjeto.MANUTENCAO_EVOLUTIVA]});
        return item;
    } 
}

